src/test/regress/sql/alter_distributed_table.sql

```diff
@@ -84,6 +84,24 @@ SELECT * FROM partitioned_table ORDER BY 1, 2;
 SELECT * FROM partitioned_table_1_5 ORDER BY 1, 2;
 SELECT * FROM partitioned_table_6_10 ORDER BY 1, 2;
 
+-- test altering partitioned table colocate_with:none
+CREATE TABLE foo (x int, y int, t timestamptz default now()) PARTITION BY RANGE (t);
+CREATE TABLE foo_1 PARTITION of foo for VALUES FROM ('2022-01-01') TO ('2022-12-31');
+CREATE TABLE foo_2 PARTITION of foo for VALUES FROM ('2023-01-01') TO ('2023-12-31');
+
+SELECT create_distributed_table('foo','x');
+
+CREATE TABLE foo_bar (x int, y int, t timestamptz default now()) PARTITION BY RANGE (t);
+CREATE TABLE foo_bar_1 PARTITION of foo_bar for VALUES FROM ('2022-01-01') TO ('2022-12-31');
+CREATE TABLE foo_bar_2 PARTITION of foo_bar for VALUES FROM ('2023-01-01') TO ('2023-12-31');
+
+SELECT create_distributed_table('foo_bar','x');
+
+SELECT COUNT(DISTINCT colocationid) FROM pg_dist_partition WHERE logicalrelid::regclass::text LIKE 'foo%';
+
+SELECT alter_distributed_table('foo', colocate_with => 'none');
+
+SELECT COUNT(DISTINCT colocationid) FROM pg_dist_partition WHERE logicalrelid::regclass::text LIKE 'foo%';
 
```

Comment:
As we discussed before, let's add such a test to make sure that everything is fine when we're cascading into the relations that the relation has foreign key relationship with too:

```sql
CREATE TABLE foo_with_fkey (x int, y int, t timestamptz default now()) PARTITION BY RANGE (t);
CREATE TABLE foo_with_fkey_1 PARTITION of foo_with_fkey for VALUES FROM ('2022-01-01') TO ('2022-12-31');
CREATE TABLE foo_with_fkey_2 PARTITION of foo_with_fkey for VALUES FROM ('2023-01-01') TO ('2023-12-31');

SELECT create_distributed_table('foo_with_fkey','x');

CREATE TABLE foo_bar_with_fkey (x int, y int, t timestamptz default now(), PRIMARY KEY (x, t)) PARTITION BY RANGE (t);
CREATE TABLE foo_bar_with_fkey_1 PARTITION of foo_bar_with_fkey for VALUES FROM ('2022-01-01') TO ('2022-12-31');
CREATE TABLE foo_bar_with_fkey_2 PARTITION of foo_bar_with_fkey for VALUES FROM ('2023-01-01') TO ('2023-12-31');

SELECT create_distributed_table('foo_bar_with_fkey','x');

ALTER TABLE foo_with_fkey ADD CONSTRAINT fkey FOREIGN KEY (x, t) REFERENCES foo_bar_with_fkey (x, t);

SELECT COUNT(DISTINCT colocationid) FROM pg_dist_partition WHERE logicalrelid::regclass::text = 'foo_with_fkey';

SELECT alter_distributed_table('foo_with_fkey', colocate_with => 'none', cascade_via_foreign_keys => true);

SELECT COUNT(DISTINCT colocationid) FROM pg_dist_partition WHERE logicalrelid::regclass::text = 'foo_bar_with_fkey';
```

Reply:
ah this doesn't make sense since we don't accept such a parameter to alter_distributed_table()